src/main/java/org/example/Main.java

```diff
@@ -1,7 +1,167 @@
 package org.example;
 
+import java.util.Scanner;
+import java.util.Set;
+
 public class Main {
     public static void main(String[] args) {
+        menu();
+    }
+
+    public static void menu() {
+        char[][] matrix = createEmptyDeck();
+
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Input command:");
+        String input = sc.nextLine();
+        String[] strings = input.split(" ");
+
+        if(input.equals("exit")) System.exit(0);
+        if(strings.length != 3) {
+            System.out.println("Bad parameters!");
+            menu();
+        }
+
+        Player player1;
+        Player player2;
+
+        if(strings[0].equals("start") &&
+                Set.of("user", "easy", "medium", "hard").contains(strings[1]) &&
+                Set.of("user", "easy", "medium", "hard").contains(strings[2])) {
+
+            player1 = switch (strings[1]) {
+                case "easy" -> new EasyAI();
+                case "user" -> new User();
+                case "medium" -> new MediumAI();
+                default -> new HardAI();
+            };
+
+            player2 = switch (strings[2]) {
+                case "easy" -> new EasyAI();
+                case "user" -> new User();
+                case "medium" -> new MediumAI();
+                default -> new HardAI();
+            };
+
+            printMatrix(matrix);
+            play(matrix, player1, player2);
+        } else {
+            System.out.println("Bad parameters!");
+            menu();
+        }
+    }
+
+    public static void play(char[][] matrix, Player player1, Player player2){
+        while(checkWinner(matrix).equals("Game not finished")) {
+            matrix = player1.play(matrix, 'X');
+            printMatrix(matrix);
+            if(checkWinner(matrix).equals("Game not finished")) {
+                matrix = player2.play(matrix, 'O');
+                printMatrix(matrix);
+                if(checkCharWinner(matrix, 'O')) {
+                    System.out.println(checkWinner(matrix));
+                    return;
+                }
+            } else {
+                System.out.println(checkWinner(matrix));
+                return;
+            }
+        }
+    }
+
+    public static String checkWinner(char[][] matrix) {
+        if (Math.abs(countXOs(matrix, 'X') - countXOs(matrix, 'O')) > 1) {
+            return "Impossible";
```

Comment:
these all return types can be an enum and used easily everywhere. Currently you are checking them with .equals() method with same strings. what if you change this return string "Impossible" to "Impossible1" and forget to change in equals? It's a good idea to have them group in a single place like enum, or some constants can also be okay but enum is better.